dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -266,15 +268,39 @@ public List<Workflow> findByPathWithoutUser(SourceControl sourceControl, String
 
     }
 
-    public List<Workflow> findPublishedByOrganizationAndRepository(SourceControl sourceControl, String organization, String repository) {
+    /**
+     * Find published workflows by source control, org and repo.
+     *
+     * @param sourceControl
+     * @param organization
+     * @param repository
+     * @param daysSinceLastRelease - if set, returns all workflows matching the other parameters, otherwise only workflows with a GitHub
+     *                             release within the window of this parameter.
+     * @return
+     */
+    public List<Workflow> findPublishedByOrganizationAndRepository(SourceControl sourceControl, String organization, String repository,
+            Integer daysSinceLastRelease) {
+        final Timestamp timestamp = getTimestamp(daysSinceLastRelease);
         return list(namedTypedQuery("io.dockstore.webservice.core.Workflow.findPublishedByOrganizationAndRepository")
+                .setParameter("releaseDate", timestamp)
+                // I tried the JPA query "...(:releaseDate is null or workflow.lastestReleaseDate > :releaseDate)", but it doesn't work against Postgres
```

Comment:
this helps, but should also be in the workflow class because it is a counter-intutitive use of parameters